File: easy/0551_student_attendance_record_i/solution.cpp
#include <string>

class Solution {
public:
  bool checkRecord(std::string s) {
    int absentCount = 0;
    int lateCount = 0;

    for (const char &record : s) {
      if (record == 'A')
        ++absentCount;

      if (record == 'L') {
        if (lateCount == 2)
          return false;
        else
          ++lateCount;
      } else
        lateCount = 0;
    }

    return absentCount < 2;
  }
};

File: easy/0551_student_attendance_record_i/solution.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        absent_count = 0
        late_count = 0

        for record in s:
            if record == "A":
                absent_count += 1

            if record == "L":
                if late_count == 2:
                    return False
                else:
                    late_count += 1
            else:
                late_count = 0

        return absent_count < 2
